### DISCUSSION TITLE
Deep Markdown Rendering

### DISCUSSION THREAD

**Discussion by d4hines** [mdx-js/mdx#1268] [Q&A] [answered]
Hi!
I'm excited to start using MDX (via [Docusaurus](https://docusaurus.io/)!). It would be nice if I could use React components for layout, and Markdown for controlling the text, and if this worked in a "deep" way. Here's an example:

```mdx
# Blah
<MySpecialLayout>
Here's some more content.

_But i can use Markdown inside JSX tags_

Useful for:
- Applying layout
- styles
- etc.
</MySpecialLayout>
```

Is there a way to do this? It doesn't work out of the box for the live editor on the site, but maybe there's a common pattern for doing what I'm trying to do.

**Comment by ChristianMurphy** [comment, marked as answer]
That is supported, it needs an extra newline between the JSX and the inner markdown

```mdx
# Blah

<MySpecialLayout>

Here's some more content.

_But i can use Markdown inside JSX tags_

Useful for:
- Applying layout
- styles
- etc.

</MySpecialLayout>
```

you can try it out at https://mdxjs.com/playground

Support further improves in MDX 2 https://github.com/mdx-js/mdx/issues/1041

**Reply by d4hines** [reply]
Wow, that is great news! Can't wait to start!
